jbpm-designer-client/src/main/java/org/jbpm/designer/client/handlers/NewCaseDefinitionHandler.java

```diff
@@ -0,0 +1,98 @@
+*/
+
+package org.jbpm.designer.client.handlers;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.google.gwt.user.client.ui.IsWidget;
+import org.guvnor.common.services.project.model.Package;
+import org.gwtbootstrap3.client.ui.Label;
+import org.gwtbootstrap3.client.ui.TextBox;
+import org.gwtbootstrap3.client.ui.constants.LabelType;
+import org.gwtbootstrap3.client.ui.gwt.HTMLPanel;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.jbpm.designer.client.resources.i18n.DesignerEditorConstants;
+import org.jbpm.designer.client.type.Bpmn2Type;
+import org.jbpm.designer.service.DesignerAssetService;
+import org.kie.workbench.common.widgets.client.handlers.DefaultNewResourceHandler;
+import org.kie.workbench.common.widgets.client.handlers.NewResourcePresenter;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.client.mvp.PlaceManager;
+import org.uberfire.commons.data.Pair;
+import org.uberfire.ext.widgets.common.client.callbacks.DefaultErrorCallback;
+import org.uberfire.mvp.PlaceRequest;
+import org.uberfire.mvp.impl.PathPlaceRequest;
+import org.uberfire.workbench.type.ResourceTypeDefinition;
+
+@ApplicationScoped
+public class NewCaseDefinitionHandler extends DefaultNewResourceHandler {
+
+    @Inject
+    private Caller<DesignerAssetService> designerAssetService;
+
+    @Inject
+    private PlaceManager placeManager;
+
+    @Inject
+    private Bpmn2Type resourceType;
+
+    private TextBox caseIdPrefixTextBox = new TextBox();
+
+    @Override
+    public String getDescription() {
+        return DesignerEditorConstants.INSTANCE.CaseDefinition();
+    }
+
+    @Override
+    public IsWidget getIcon() {
+        return null;
+    }
+
+    @Override
+    public ResourceTypeDefinition getResourceType() {
+        return resourceType;
+    }
+
+    @PostConstruct
+    private void setupExtensions() {
+        extensions.add( Pair.newPair( "CaseIdPrefixLabel", new HTMLPanel(DesignerEditorConstants.INSTANCE.CaseIdPrefix())) );
+        extensions.add( Pair.newPair( "CaseIdPrefixTextBox", caseIdPrefixTextBox ) );
+    }
+
+    @Override
+    public void create( final Package pkg,
+                        final String baseFileName,
+                        final NewResourcePresenter presenter ) {
+        String caseIdPrefix = caseIdPrefixTextBox.getValue();
+        caseIdPrefixTextBox.clear();
+
+        designerAssetService.call( new RemoteCallback<Path>() {
+            @Override
+            public void callback( final Path path ) {
+                presenter.complete();
+
+                notifySuccess();
+                final PlaceRequest place = new PathPlaceRequest( path );
+                placeManager.goTo( place );
+            }
+        }, new DefaultErrorCallback() ).createCaseDefinition( pkg.getPackageMainResourcesPath(), buildFileName( baseFileName,
```

Comment:
I know this has been merged already, but just for future PRs: using `new DefaultErrorCallback()` should no longer be necessary as it is used as default uncaught exception handler (see [here](https://github.com/droolsjbpm/kie-wb-common/pull/462/files#diff-a16fcaf1425005de11fd05cffed014c7R65))